cub/test/catch2_test_device_select_flagged_if.cu

```diff
@@ -0,0 +1,283 @@
+
+using flag_types = c2h::type_list<std::uint8_t, std::uint64_t, custom_t>;
+
+CUB_TEST("DeviceSelect::FlaggedIf can run with empty input", "[device][select_flagged_if]", types)
+{
+  using type = typename c2h::get<0, TestType>;
+
+  constexpr int num_items = 0;
+  c2h::device_vector<type> in(num_items);
+  c2h::device_vector<type> out(num_items);
+  c2h::device_vector<int> flags(num_items);
+
+  // Needs to be device accessible
+  c2h::device_vector<int> num_selected_out(1, 0);
+  int* d_num_selected_out = thrust::raw_pointer_cast(num_selected_out.data());
+
+  select_flagged_if(in.begin(), flags.begin(), out.begin(), d_num_selected_out, num_items, always_true_t{});
+
+  REQUIRE(num_selected_out[0] == 0);
+}
+
+CUB_TEST("DeviceSelect::FlaggedIf handles all matched", "[device][select_flagged_if]", types)
+{
+  using type = typename c2h::get<0, TestType>;
+
+  const int num_items = GENERATE_COPY(take(2, random(1, 1000000)));
+  c2h::device_vector<type> in(num_items);
+  c2h::device_vector<type> out(num_items);
+  c2h::device_vector<int> flags(num_items);
+  c2h::gen(CUB_SEED(2), in);
+
+  // Needs to be device accessible
+  c2h::device_vector<int> num_selected_out(1, 0);
+  int* d_first_num_selected_out = thrust::raw_pointer_cast(num_selected_out.data());
+
+  select_flagged_if(in.begin(), flags.begin(), out.begin(), d_first_num_selected_out, num_items, always_true_t{});
+
+  REQUIRE(num_selected_out[0] == num_items);
+  REQUIRE(out == in);
+}
+
+CUB_TEST("DeviceSelect::FlaggedIf handles no matched", "[device][select_flagged_if]", types)
+{
+  using type = typename c2h::get<0, TestType>;
+
+  const int num_items = GENERATE_COPY(take(2, random(1, 1000000)));
+  c2h::device_vector<type> in(num_items);
+  c2h::device_vector<type> out(0);
+  c2h::gen(CUB_SEED(2), in);
+
+  c2h::device_vector<int> flags(num_items, 0);
+
+  // Needs to be device accessible
+  c2h::device_vector<int> num_selected_out(1, 0);
+  int* d_first_num_selected_out = thrust::raw_pointer_cast(num_selected_out.data());
+
+  select_flagged_if(in.begin(), flags.begin(), out.begin(), d_first_num_selected_out, num_items, always_false_t{});
+
+  REQUIRE(num_selected_out[0] == 0);
+}
+
+CUB_TEST("DeviceSelect::FlaggedIf does not change input and is stable",
+         "[device][select_flagged_if]",
+         c2h::type_list<std::uint8_t, std::uint64_t>,
+         flag_types)
+{
+  using input_type = typename c2h::get<0, TestType>;
+  using flag_type  = typename c2h::get<1, TestType>;
+
+  const int num_items = GENERATE_COPY(take(2, random(1, 1000000)));
+  c2h::device_vector<input_type> in(num_items);
+  c2h::device_vector<input_type> out(num_items);
+  c2h::gen(CUB_SEED(2), in);
+
+  is_even_t<flag_type> is_even{};
+
+  c2h::device_vector<flag_type> flags(num_items);
+  c2h::gen(CUB_SEED(1), flags);
+  const c2h::host_vector<input_type> reference_out = get_reference(in, flags, is_even);
+  const std::size_t num_selected                   = reference_out.size();
```

Comment:
In `REQUIRE(num_selected == num_selected_out[0]);`, we are now comparing `int` vs. `size_t`, i.e., `signed/unsigned mismatch` in comparison. Something our MSVC build often complains about. In this case, it didn't. Maybe for [this reason](https://stackoverflow.com/questions/5416414/why-does-msvc-not-issue-a-warning-for-signed-unsigned-comparison)(?)
Nonetheless, I think it's better to fix it by either (a) casting the reference num_selected to `int` or (b) casting in the comparison.
```suggestion
  const int num_selected                   = static_cast<int>(reference_out.size());
```
Same applies to the two tests below.

Reply:
agreed. fixed.